lib/translate/inquiry.cc

```diff
@@ -0,0 +1,318 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "inquiry.h"
+#include "string.h"
+
+namespace translator {
+
+// Creates and validates a Inquiry Command struct
+StatusCode RawToScsiCommand(absl::Span<const uint8_t> raw_cmd,
+                            scsi_defs::InquiryCommand &cmd) {
+  if (raw_cmd.empty()) {
+    char debug_buffer[100];
+    sprintf(debug_buffer, "buffer is empty or nullptr\n");
+    if (debug_callback != NULL) debug_callback(debug_buffer);
+    return StatusCode::kInvalidInput;
+  }
+
+  scsi_defs::OpCode opcode;
+  StatusCode opcode_status = MakeScsiOpcode(raw_cmd[0], opcode);
+  if (opcode_status != StatusCode::kSuccess) {
+    return opcode_status;
+  }
+  if (opcode != scsi_defs::OpCode::kInquiry) {
+    char debug_buffer[100];
+    const char *expected_cmd_str =
+        ScsiOpcodeToString(scsi_defs::OpCode::kInquiry);
+    const char *cmd_str = ScsiOpcodeToString(opcode);
+    sprintf(debug_buffer, "invalid opcode. expected %s got %s.",
+            expected_cmd_str, cmd_str);
+    if (debug_callback != NULL) debug_callback(debug_buffer);
+
+    return StatusCode::kInvalidInput;
+  }
+
+  // TODO: check invalid parameters in scsi_command
+  cmd = *reinterpret_cast<const scsi_defs::InquiryCommand *>(&raw_cmd[1]);
+  return StatusCode::kSuccess;
+}
+
+// TODO: Executes the NVME Identify Controller command
+nvme_defs::IdentifyControllerData NvmeIdentifyController() {
+  return nvme_defs::IdentifyControllerData();
+}
+
+// TODO: Executes the NVME Identify Namespace command
+nvme_defs::IdentifyNamespace NvmeIdentifyNamespace() {
```

Comment:
We should remove the execution functions from this PR as they are out of scope of this file

Reply:
we will reorganize these in a different pr

Reply:
Unresolving as I think these functions, in their current state, don't need to go into master as they serve no real purpose. @eyhhuang @basimsahaf thoughts?

Reply:
I agree, esp since under the updated design, the translation lib does not handle any of the nvme execution